src/OpenSage.Game/Logic/Object/GameObject.cs

```diff
@@ -539,7 +511,7 @@ internal void FinishConstruction()
             {
                 if (behavior is SpawnBehaviorModuleData spawnBehaviorModuleData)
                 {
-                    Spawn(spawnBehaviorModuleData.SpawnTemplate.Value);
+                    ProductionUpdate.Spawn(spawnBehaviorModuleData.SpawnTemplate.Value);
```

Comment:
@feliwir noted that this will only work for objects with a ProductionUpdate behavior. I will implement the SpawnBehavior in a sperate PR and fix this properly.